### PR TITLE
debut design v2

### PR DESCRIPTION
**Pull request by Octopus773**
ajout du design de la navigation
+ homeview

### CODE REVIEW

File: front/views/V2/HomeView.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+const bigSideRatio = 1000;
+const smallSideRatio = 618;
+
+type HomeCardProps = {
+	image: string;
+	title: string;
+	artist: string;
+};
+
+const cards = [
```
Comment (Arthi-chaud) [inline front/views/V2/HomeView.tsx:18]: No hard-coded data anymore. Please

File: front/views/V2/HomeView.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+						fontSize: 24,
+						fontWeight: 'bold',
+						marginLeft: 16,
+						marginBottom: 16,
+						marginTop: 24,
+					}}
+				>
+					{'Suggestions'}
+				</Text>
+				{songsQuery.isLoading && <Text>Loading...</Text>}
```
Comment (Arthi-chaud) [inline front/views/V2/HomeView.tsx:150]: This should be Translated

File: front/components/V2/TabNavigation.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	isActive?: boolean;
+	isCollapsed?: boolean;
+	iconName?: string;
+};
+
+const tabs = [
+	{
+		id: 'home',
+		label: 'Discovery',
+		icon: <Ionicons name="search" size={24} color="black" />,
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigation.tsx:25]: The icons should be in white

File: front/components/V2/TabNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+	const appTabs = tabs.map((t) => {
+		// use the same instance of a component between desktop and mobile
+		return {
+			...t,
+			onPress: () => setActiveTab(t.id),
+			icon: (
+				<Ionicons
+					// eslint-disable-next-line @typescript-eslint/no-explicit-any
+					name={t.iconName as any}
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigation.tsx:76]: Use of type-casting should be explained

File: front/components/V2/SongCardInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+							<Button
+								onHoverIn={() => {
+									setIsPlayHovered(true);
+								}}
+								onHoverOut={() => {
+									setIsPlayHovered(false);
+								}}
+								borderRadius={100}
+								marginBottom={35}
+								leftIcon={
```
Comment (Arthi-chaud) [inline front/components/V2/SongCardInfo.tsx:236]: This makes the icon to be slightly off-centered to the left. This should  be fixed

File: front/components/V2/TabNavigation.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+		icon: <Ionicons name="notifications" size={24} color="black" />,
+		iconName: 'notifications',
+	},
+	{
+		id: 'settings',
+		label: 'Settings',
+		icon: <Ionicons name="settings" size={24} color="white" />,
+		iconName: 'settings',
+	},
+] as NaviTab[];
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigation.tsx:58]: use `as const` instead

File: front/views/V2/HomeView.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+		image: 'https://media.discordapp.net/attachments/717080637038788731/1153688154499457096/image_homeview3.png',
+		title: 'Back',
+		artist: 'Truc Truc',
+	},
+	{
+		image: 'https://media.discordapp.net/attachments/717080637038788731/1153688154109394985/image_homeview4.png',
+		title: 'Mozart',
+		artist: 'Machin Machin',
+	},
+] as [HomeCardProps, HomeCardProps, HomeCardProps, HomeCardProps];
```
Comment (Arthi-chaud) [inline front/views/V2/HomeView.tsx:39]: use `as const` instead

File: front/components/V2/TabNavigationPhone.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+import { NaviTab } from './TabNavigation';
+
+type TabNavigationPhoneProps = {
+	tabs: NaviTab[];
+	activeTabID: string;
+	setActiveTabID: (id: string) => void;
+	children?: React.ReactNode;
+};
+
+const TabNavigationPhone = (props: TabNavigationPhoneProps) => {
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigationPhone.tsx:13]: The name of this component is misleading. It should be either renamed or documented

File: front/components/V2/TabNavigationButton.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+							if (isHovered) {
+								return '0px 0px 16px 0px rgba(0, 0, 0, 0.25)';
+							} else if (props.isActive) {
+								return '0px 0px 8px 0px rgba(0, 0, 0, 0.25)';
+							} else {
+								return undefined;
+							}
+						})(),
+						backdropFilter: 'blur(2px)',
+						backgroundColor: (() => {
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigationButton.tsx:45]: Couldn't these values be used elsewhere? Shouldn't they be in the global theme? Especially if they are raw CSS

File: front/components/V2/TabNavigationDesktop.tsx [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+	setIsCollapsed: (isCollapsed: boolean) => void;
+	activeTabID: string;
+	setActiveTabID: (id: string) => void;
+	children?: React.ReactNode;
+};
+
+const TabNavigationDesktop = (props: TabNavigationDesktopProps) => {
+	const colorScheme = useColorScheme();
+	const [icon] = useAssets(
+		colorScheme == 'light'
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigationDesktop.tsx:21]: With the current color scheme, there is no need to handle light and dark modes

File: front/components/V2/TabNavigationDesktop.tsx [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+			? require('../../assets/icon_light.png')
+			: require('../../assets/icon_dark.png')
+	);
+	// settings is displayed separately (with logout)
+	const buttons = props.tabs.filter((tab) => tab.id !== 'settings');
+
+	const others = [
```
Comment (Arthi-chaud) [inline front/components/V2/TabNavigationDesktop.tsx:28]: No hard coded data please

**Review by Arthi-chaud — APPROVED** [review]
**DID NOT MEAN TO APPROVE**

**Review by Arthi-chaud — CHANGES_REQUESTED** [review]
Almost no Buttons have callbacks.
I don't think this should be merged before the implementation is complete.

Also, fix this:
Screenshot 2023-09-20 at 09 36 03

File: front/components/V2/TabNavigation.tsx (hunk 4 above)
Comment (Octopus773) [reply front/components/V2/TabNavigation.tsx:76]: J'ai pas trouvé le type de l'enum ionicons

File: front/components/V2/TabNavigationButton.tsx (hunk 9 above)
Comment (Octopus773) [reply front/components/V2/TabNavigationButton.tsx:45]: Actuellement non Mathys à un component pour faire ça cool mais par manque de temps j'applique directement les styles de boxshadow en fonction d'un scénario 

Dans l'idée oui, dans cette pr non

**Pull request merged**